applications/dashboard/models/class.usermodel.php

```diff
@@ -876,15 +887,16 @@ public function connect($uniqueID, $providerKey, $userData, $options = []) {
                 $userData['Password'] = md5(microtime());
                 $userData['HashMethod'] = 'Random';
 
-                touchValue('CheckCaptcha', $options, false);
-                touchValue('NoConfirmEmail', $options, true);
-                touchValue('NoActivity', $options, true);
-
                 // Translate SSO style roles to an array of role IDs suitable for registration.
                 if (!empty($userData['Roles']) && !isset($userData['RoleID'])) {
                     $userData['RoleID'] = $this->lookupRoleIDs($userData['Roles']);
                 }
-                touchValue('SaveRoles', $options, !empty($userData['RoleID']) && c('Garden.SSO.SyncRoles', false));
+
+                $options['CheckCaptcha'] = isset($options['CheckCaptcha']) ? $options['CheckCaptcha'] : false;
```

Comment:
> This would be a perfect fit for the [null coalescing operator](http://php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op). I think there are a few other candidates for null-coalescing in here.

Reply:
Think I got all of 'em now.